### PR TITLE
Job Portal Importer

### PR DESCRIPTION
**Pull request by davdenic**
first PR, porting of the PoC into vertical_job_portal

### CODE REVIEW

**Comment by janikvonrotz** [comment]
I ran the pylint with the odoolint module:

```
************* Module job_portal_importer.models.transformation
addons/vertical_job_portal/job_portal_importer/models/transformation.py:7:11: W8113: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
addons/vertical_job_portal/job_portal_importer/models/transformation.py:9:12: W8113: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
************* Module job_portal_importer.models.validation
addons/vertical_job_portal/job_portal_importer/models/validation.py:7:11: W8113: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
addons/vertical_job_portal/job_portal_importer/models/validation.py:9:12: W8113: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
************* Module job_portal_importer.models.provider
addons/vertical_job_portal/job_portal_importer/models/provider.py:7:13: W8113: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
addons/vertical_job_portal/job_portal_importer/models/provider.py:10:12: W8113: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
************* Module job_portal_importer.models.contract
addons/vertical_job_portal/job_portal_importer/models/contract.py:7:13: W8113: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
addons/vertical_job_portal/job_portal_importer/models/contract.py:11:12: W8113: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)

------------------------------------------------------------------
Your code has been rated at 8.45/10 (previous run: 8.45/10, +0.00)

Run pre-commit in addons/vertical_job_portal/job_portal_importer
An error has occurred: InvalidConfigError:
=====> .pre-commit-config.yaml is not a file
Check the log at /home/janikvonrotz/.cache/pre-commit/pre-commit.log
Update index.html
README.rst:5: (WARNING/2) Title overline too short.

===============
Job Portal Importer
===============
```

Can you fix these issues as well?

**Comment by janikvonrotz** [comment]
@davdenic  Or even better, can you create a github action that runs the linter https://github.com/OCA/pylint-odoo using https://pre-commit.com ?

**Comment by davdenic** [comment]
@janikvonrotz now everything seems ok after your changes
```
forbidden files......................................(no files to check)Skipped
en.po files cannot exist.............................(no files to check)Skipped
autoflake................................................................Passed
black....................................................................Passed
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
debug statements (python)................................................Passed
fix python encoding pragma...............................................Passed
check for case conflicts.................................................Passed
check docstring is first.................................................Passed
check that executables have shebangs.................(no files to check)Skipped
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
check xml................................................................Passed
mixed line ending........................................................Passed
isort except __init__.py.................................................Passed
flake8...................................................................Passed
pylint with optional checks..............................................Passed
```

File: job_portal_importer/models/contract.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from odoo import api, fields, models
+
+class ImporterContract(models.Model):
+    _name = "importer.contract"
```
[the remark this answers is not in the document]
Comment (davdenic) [reply job_portal_importer/models/contract.py:5]: if I do those changes then I have the problem like

odoo.exceptions.ValidationError: Table name 'job_portal_importer_mapping_job_portal_importer_mapping_rule_rel' is too long

File: job_portal_importer/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+    'name': 'Sozialinfo Job importer',
+    "summary": """
+         Job portal importer via XML.
+    """,
+    'author': 'David Denicolo',
+    "category": "Services",
+    "version": "16.0.1.0.0",
+    "license": "AGPL-3",
+    "depends": ["base", "job_portal_base"],
```
[the remark this answers is not in the document]
Comment (davdenic) [reply job_portal_importer/__manifest__.py]: done

File: job_portal_importer/static/description/icon.png
[the remark this answers is not in the document]
Comment (davdenic) [reply job_portal_importer/static/description/icon.png:1]: done

[the remark this answers is not in the document]
Comment (davdenic) [reply job_portal_importer/static/description/icon.png:1]: as soon as we are ready sure but for the moment i guess the sozialinfo icon i've used seems ok to me

File: job_portal_importer/models/contract.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (davdenic) [reply job_portal_importer/models/contract.py]: done

File: job_portal_importer/security/ir.model.access.csv
Comment (janikvonrotz) [inline job_portal_importer/security/ir.model.access.csv:1]: A manager group is introduced with the job_portal_base module. Can you ass the permissions to this group?

Comment (janikvonrotz) [inline job_portal_importer/security/ir.model.access.csv:1]: Can you use the group_id from job_portal_base?

Comment (janikvonrotz) [reply job_portal_importer/security/ir.model.access.csv:1]: @davdenic 
The id is: job_portal_base.group_job_portal_joboffer_manager

**Review by janikvonrotz — APPROVED** [review]

**Pull request merged**